packages/graphql/src/server.js

```diff
@@ -15,7 +15,52 @@ const schema = makeExecutableSchema({
   resolverValidationOptions: { requireResolversForResolveType: false }
 })
 
-const server = new ApolloServer({ schema })
+const options = {
+  schema,
+  context: ({ req }) => {
+    const operation = req.body
+    if (operation && operation.query) {
+      /**
+       * TODO: Maybe check how apollo parses queries and use the same parser.
+       * There could be a case here where a character is inserted before
+       * 'motation' in the query body that apollo would ignore, but would cause
+       * this to fail.  While I don't think any of the mutations being run from
+       * the server are *dangerous*, it sure isn't ideal and could have some
+       * unintended side-effects.
+       */
+      const match = operation.query.match(/^\s*(mutation)/)
+      if (match) {
+        console.warn(
+          `Mutations not allowed.  Operation: ${operation.operationName}`
+        )
+        throw new Error('Mutations not allowed')
+      }
+    }
+  }
+}
+
+const API_KEY = process.env.ENGINE_API_KEY || process.env.APOLLO_METRICS_API_KEY
```

Comment:
Why using 2 env vars ?

Reply:
`ENGINE_API_KEY` is supported by ApolloServer and is what's in their docs.  I preferred `APOLLO_METRICS_API_KEY` because it's purpose is clearer.  I thought it best to support the one in the docs in case someone uses it.  If I didn't add support for `ENGINE_API_KEY` and someone used it, it might cause some unexpected behavior that I thought best to avoid.

Reply:
Ok. Maybe add a comment.